File: src/main/java/kingdoms/server/handlers/packets/client/CPacketSyncBank.java
package kingdoms.server.handlers.packets.client;

import kingdoms.api.network.AbstractPacket;
import kingdoms.server.PlayerProvider;
import net.minecraft.entity.player.EntityPlayer;

public final class CPacketSyncBank extends AbstractPacket
{
    public CPacketSyncBank(){}
    public CPacketSyncBank(int count, boolean bank)
    {
        buf().writeInt(count);
        buf().writeBoolean(bank);
    }

    @Override
    public void client(EntityPlayer player)
    {
        PlayerProvider provider = PlayerProvider.get(player);

        int count = buf().readInt();
        boolean bank = buf().readBoolean();

        if (bank)
        {
            provider.setBankGold(count);
        }
        else {
            provider.setGoldTotal(count);
        }
    }
}

File: src/main/java/kingdoms/server/handlers/packets/client/CPacketSyncBank.kt
package kingdoms.server.handlers.packets.client

import kingdoms.api.network.SimplePacket
import kingdoms.server.PlayerProvider
import net.minecraft.entity.player.EntityPlayer

class CPacketSyncBank(): SimplePacket()
{
    constructor(count: Int, bank: Boolean): this()
    {
        buf().writeInt(count)
        buf().writeBoolean(bank)
    }

    override fun client(player: EntityPlayer)
    {
        val provider = PlayerProvider.get(player)

        val count = buf().readInt()
        val bank = buf().readBoolean()

        if (bank)
        {
            provider.bankGold = count
        }
        else
        {
            provider.goldTotal = count
        }
    }
}